src/caret_analyze/architecture/architecture_loaded.py

```diff
@@ -766,22 +780,58 @@ def __init__(
         callbacks_loaded: CallbacksLoaded,
         node: NodeValue
     ) -> None:
+        # duplicate check for callback_id
         subscription_values = reader.get_subscriptions(node)
+        try:
+            self._validate(subscription_values)
+        except InvalidReaderError as e:
+            logger.warn(e)
+
+        subscription_values = self._remove_duplicated(subscription_values)
+
         self._data = [self._to_struct(callbacks_loaded, sub)
                       for sub in subscription_values]
 
+    @staticmethod
+    def _validate(callback_vals: Sequence[SubscriptionValue]):
+        # validate callback id uniqueness.
+        callback_ids: Set[str] = set()
+        duplicated: List[str] = []
+        for callback_id in callback_vals:
+            if callback_id.callback_id is not None:
+                if callback_id.callback_id not in callback_ids:
+                    callback_ids.add(callback_id.callback_id)
+                else:
+                    duplicated.append(callback_id.callback_id)
+        if len(callback_ids) != len(callback_vals):
```

Comment:
If there is an instance with callback_id of None, a duplicated callback id exception occurs.
This is not the intended behaviour and needs to be fixed.
```suggestion
        if len(duplicated) > 0:
```

Reply:
Fixed.